iconfetcher.go

```diff
@@ -0,0 +1,108 @@
+package jujusvg
+
+import (
+	"io/ioutil"
+	"net/http"
+	"sync"
+
+	"gopkg.in/errgo.v1"
+	"gopkg.in/juju/charm.v5-unstable"
+)
+
+// An IconFetcher provides functionality for retrieving icons for the charms
+// within a given bundle.  The FetchIcons function accepts a bundle, and
+// returns a dict mapping charm paths to icon data.
+type IconFetcher interface {
+	FetchIcons(*charm.BundleData) (map[string]string, error)
+}
+
+// HttpFetcher is a default implementation of IconFetcher which retrieves charm
```

Comment:
s/a default/an/

(the default behaviour is documented elsewhere and isn't directly relevant here, I think)